### ISSUE TITLE
[BUG]: SocketException: Resource temporarily unavailable

### ISSUE THREAD

**Issue by ooples** [alpacahq/alpaca-trade-api-csharp#522]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

```cs
Exception Type:
System.Net.Sockets.SocketException

Exception Message:
Resource temporarily unavailable

Exception StackTrace:
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16 token)
   at System.Net.Sockets.Socket.<ConnectAsync>g__WaitForConnectWithCancellation|277_0(AwaitableSocketAsyncEventArgs saea, ValueTask connectTask, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.ConnectToTcpHostAsync(String host, Int32 port, HttpRequestMessage initialRequest, Boolean async, CancellationToken cancellationToken)
   at Polly.Retry.AsyncRetryEngine.ImplementationAsync[TResult](Func`3 action, Context context, CancellationToken cancellationToken, ExceptionPredicates shouldRetryExceptionPredicates, ResultPredicates`1 shouldRetryResultPredicates, Func`5 onRetryAsync, Int32 permittedRetryCount, IEnumerable`1 sleepDurationsEnumerable, Func`4 sleepDurationProvider, Boolean continueOnCapturedContext)
   at Polly.AsyncPolicy.ExecuteAsync[TResult](Func`3 action, Context context, CancellationToken cancellationToken, Boolean continueOnCapturedContext)
   at Polly.Wrap.AsyncPolicyWrapEngine.ImplementationAsync[TResult](Func`3 func, Context context, CancellationToken cancellationToken, Boolean continueOnCapturedContext, IAsyncPolicy outerPolicy, IAsyncPolicy`1 innerPolicy)
   at Polly.AsyncPolicy`1.ExecuteAsync(Func`3 action, Context context, CancellationToken cancellationToken, Boolean continueOnCapturedContext)
   at Alpaca.Markets.ThrottleParameters.CustomHttpHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposeCts, CancellationTokenSource pendingRequestsCts, CancellationToken originalCancellationToken)
   at Alpaca.Markets.HttpClientExtensions.callAndDeserializeAsync[TApi,TJson](HttpClient httpClient, HttpRequestMessage request, CancellationToken cancellationToken)
   at Alpaca.Markets.HttpClientExtensions.callAndDeserializeAsync[TApi,TJson](HttpClient httpClient, HttpMethod method, Uri endpointUri, CancellationToken cancellationToken)
   at Alpaca.Markets.AlpacaTradingClient.ListOrdersAsync(ListOrdersRequest request, CancellationToken cancellationToken)
   at OoplesFinanceAdminClient.AlpacaApi.SendDailyReportEmail() in E:\OoplesFinanceAdminClient\OoplesFinanceAdminClient\AlpacaApi.cs:line 750

Exception Target Site:
ThrowException

Exception Source:
System.Net.Sockets
```

### Steps To Reproduce

This is definitely a new one for me and as far as I have seen it only happened on the same computer (Ubuntu 20.04) for all alpaca clients I had running at the time. This happened for all clients around the same time and as far as I can tell, it didn't try to reconnect after this error happened.

### Environment

- SDK Version: 6.0.4
- OS (version, bitness): Ubuntu 20.04 x64
- .NET SDK (version): 6.0
- target process .NET version/bitness: 6.0 x64

**Comment by OlegRa** [comment]
@ooples I'm unable to reproduce this problem on my side. After checking the .NET sources I can confirm that this error can be generated by the Linux implementation of the socket wrapper. But I'm not sure we need any changes in SDK code:

1. Not all socket errors are recoverable and the retry policy shouldn't treat any error as a signal for reconnecting. Some errors should be handled on the caller's side.
2. If you are sure that this error is recoverable you can add a related socket error code to the list of retry socket codes - `ThrottleParameters.RetrySocketErrorCodes` property created specifically for this case.

In any case, if you can reproduce it and provide more information about the exception (the `ErrorCode` and `SocketErrorCode` properties values) it would be also helpful. I'll not close this issue right now because this additional data can change my opinion about the "recoverability" of this error.

**Comment by ooples** [comment]
@OlegRa Socket error code is WOULDBLOCK and error code is 10035

**Comment by OlegRa** [comment]
I've checked this error code description and code conditions that can generate it and yes, it looks like a "recoverable" socket error. Most probably the root cause of this error is an overloaded network output buffer on client or receive buffer on the server. The random delay can help so at least we can try. But I'm not sure this fix is urgent - I will make the fix without publishing the release immediately. As a temporary workaround, you can add the `SocketError.WouldBlock` error code into the `ThrottleParameters.RetrySocketErrorCodes` set manually.

**Comment by OlegRa** [comment]
@ooples You can test this fix in version [6.0.5](/releases/tag/sdk-6.0.5) of the SDK package. If it's hard to reproduce - just close it and we'll re-open it later if the problem is not fixed.
